src/sentry/workflow_engine/endpoints/project_detector_index.py

```diff
@@ -20,6 +20,26 @@
 from sentry.workflow_engine.models import Detector
 
 
+def get_validator(request, project, detector_type_slug, instance=None):
```

Comment:
this is just so that it can be reused, kinda unrelated to the rest of the changes but makes my [PUT endpoint PR](https://github.com/getsentry/sentry/pull/84098) smaller

Reply:
```suggestion
def get_detector_validator(request: Request, project: Project, detector_type_slug: str, instance=None):
```
i think it'd help to update the name a little and set the types. maybe this could also set the return type?